src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,195 @@
+            "dd-MM-yyyy HH:mm");
+
+    private static LocalDateTime parseInputString(String inputDateTime)
+            throws DateTimeParseException {
+        return LocalDateTime.parse(inputDateTime, formatter);
+    }
+
+    /**
+     * Returns a boolean corresponding to whether the given string is numeric.
+     *
+     * @param strNum
+     *            String to be tested
+     * @return boolean representing whether the string is numeric
+     */
+    private static boolean isNumeric(String strNum) {
+        if (strNum == null) {
+            return false;
+        }
+        try {
+            Integer.parseInt(strNum);
+        } catch (NumberFormatException nfe) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * The parse function takes a string as input and parses it into commands.
+     * The parse function is responsible for parsing the user's input into
+     * commands that can be executed by the program. It also handles any errors
+     * that may occur during parsing, such as missing arguments or invalid date/time
+     * formats.
+     *
+     * @param userInput
+     *            Get the user input
+     */
+    public BaseCommand parse(String userInput)
+            throws MissingArgumentException, InvalidDateTimeException, InvalidTaskSpecificationException,
+            IncorrectArgumentException, UnknownCommandException, NoCommandException {
+        if (userInput.trim().isEmpty()) {
+            throw new NoCommandException(
+                    "Please enter a command");
+        }
+
+        String[] commandArgs = userInput.split(" ");
+        String[] commandArgsCopy = new String[commandArgs.length - 1];
+        System.arraycopy(
+                commandArgs,
+                1,
+                commandArgsCopy,
+                0,
+                commandArgs.length - 1);
+
+        switch (commandArgs[0]) {
+        case exitCommand:
+            return new ExitCommand();
+        case todoCommand:
+            String todoText = String.join(" ", commandArgsCopy);
+
+            if (commandArgsCopy.length > 0) {
+                Task newTodo = Todo.of(todoText);
+                return new AddTaskCommand(newTodo);
+            } else {
+                throw new MissingArgumentException(
+                        "The description of the todo cannot be empty!");
+            }
+        case deadlineCommand:
+            String deadlineText = String.join(" ", commandArgsCopy);
+            if (deadlineText.contains(deadlineSubCommand)) {
+                String[] deadlineArgs = deadlineText.split(
+                        deadlineSubCommand);
+                String deadlineTitle = deadlineArgs[0];
+                String deadline = deadlineArgs[1];
+
+                Task newDeadline = Task.of(
+                        "D",
+                        "0",
+                        deadlineTitle,
+                        deadline);
```

Comment:
Perhaps put it in the same line to reduce the level of indentation to make it easier to read?

Reply:
Oh, I thought it was easier to read like this but I shall take note